src/main.ts

```diff
@@ -24,15 +24,31 @@ function colorize(severity: Severity, message: string): string {
 
 import { parseInput } from './input.ts';
 import { fetchChecks } from './github-api.ts';
-import { Severity, generateReport, getSummaries, readableDuration } from './report.ts';
+import { Report, Severity, generateReport, getSummaries, readableDuration } from './report.ts';
 import { getInterval, wait } from './wait.ts';
 import { Temporal } from 'temporal-polyfill';
+import { Check, Options, Trigger } from './schema.ts';
+import { WebhookPayload } from '@actions/github/lib/interfaces.ts';
+
+interface Attempt {
+  elapsed: Temporal.Duration;
+  checks: Check[];
+  report: Report;
+}
+
+interface Dumper {
+  trigger: Trigger;
+  options: Options;
+  payload: WebhookPayload;
```

Comment:
I want a payload for addressing issues, such as https://github.com/kachick/wait-other-jobs/pull/767.
However, the object schema scares me due to accidents. 🤔

Reply:
```
[github context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)
The github context contains information about the workflow run and the event that triggered the run. You can also read most of the github context data in environment variables. For more information about environment variables, see "[Variables](https://docs.github.com/en/actions/learn-github-actions/variables)."

Warning: When using the whole github context, be mindful that it includes sensitive information such as github.token. GitHub masks secrets when they are printed to the console, but you should be cautious when exporting or printing the context.

Warning: When creating workflows and actions, you should always consider whether your code might execute untrusted input from possible attackers. Certain contexts should be treated as untrusted input, as an attacker could insert their own malicious content. For more information, see "[Security hardening for GitHub Actions](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections)."
```

https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

In my understanding, this payload will points to `github.event`, not the whole of context.

```
The full event webhook payload. You can access individual properties of the event using this context. This object is identical to the webhook payload of the event that triggered the workflow run, and is different for each event. The webhooks for each GitHub Actions event is linked in "Events that trigger workflows." For example, for a workflow run triggered by the push event, this object contains the contents of the push webhook payload.
```

But...